src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorArchiveCase.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.hmcts.divorce.solicitor.event;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CCDConfig;
+import uk.gov.hmcts.ccd.sdk.api.ConfigBuilder;
+import uk.gov.hmcts.divorce.common.ccd.PageBuilder;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.model.UserRole;
+
+import static uk.gov.hmcts.divorce.divorcecase.model.State.Archived;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.PRE_SUBMISSION_STATES;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.APPLICANT_1_SOLICITOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.APPLICANT_2_SOLICITOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASE_WORKER;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
+
+@Slf4j
+@Component
+public class SolicitorArchiveCase implements CCDConfig<CaseData, State, UserRole> {
+
+    public static final String SOLICITOR_ARCHIVE_CASE = "solicitor-archive-case";
+    private static final String ARCHIVE_CASE = "Archive Case";
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+
+        new PageBuilder(configBuilder
+            .event(SOLICITOR_ARCHIVE_CASE)
+            .forStateTransition(PRE_SUBMISSION_STATES, Archived)
+            .name(ARCHIVE_CASE)
+            .description(ARCHIVE_CASE)
+            .grant(CREATE_READ_UPDATE, APPLICANT_1_SOLICITOR)
+            .grantHistoryOnly(CASE_WORKER, LEGAL_ADVISOR, SUPER_USER, APPLICANT_2_SOLICITOR)
```

Comment:
can caseworkers search the case if the state is Archived ?

Reply:
See previous comment for solution.